### ISSUE TITLE
[bug almost fixed in v2.0.4] zpm console lag

### ISSUE THREAD

**Issue by C4PTRAMPAGE** [AlmuraDev/SGCraft#156]
after some testing the bug https://github.com/AlmuraDev/SGCraft/issues/135 IS completely gone... if the console is placed facing east or west. however if placed on top of a solid block so it faces north or south the freezes still happen.

**Comment by Dockter** [comment]
That literally doesn't make sense.  This being said, can you attach some picture to this ticket with the F3 diag screen up as well.  This should help give me some context to this very strange issue.

**Comment by C4PTRAMPAGE** [comment]
yeah well its whats happening, and i can do 60 fps better than a picture https://youtu.be/0m7tzNbjKbE

**Comment by Bysokar** [comment]
+1 this for players on stargatemc.com :(

**Comment by Bysokar** [comment]
@dockter The model is faulty. I just re-imported it into Crayfish's updated editor and 'optomised it' and it found errors with some of the values.

Reimport and recompile and it works beautifully.

**Comment by Dockter** [comment]
Can you give me the model it created?

**Comment by Bysokar** [comment]
[zpm_console.zip](https://github.com/AlmuraDev/SGCraft/files/4951475/zpm_console.zip)

@Dockter Here you go. Note the reduction in filesize to start... but it also threw a bunch of errors about model verticies not being a multiple of 16.

Works a charm for me now.

**Comment by Dockter** [comment]
I am testing this internally.

**Comment by Dockter** [comment]
@draksterau my @wifee wants to know what version of model editors you're using.  She said she used crayfish editor to do the original model so she's confused as to why it would have suddenly generated errors.

**Comment by Bysokar** [comment]
mc-0.7.0 :)

I imported json, then exported. Boom problem solved.

When I imported it said there were verticies that didn't have a multiple of 16. When I exported I selected optimize model.
Sorry I don't really know what im doing regarding this... but I do know it worked, hence mentioning.

@Dockter Also, I've implemented Universe gates in the StargateMC.com build of SGCraft for 1.12. You'd be welcome to the textures and sounds if you want to implement it. Code too...

**Comment by Dockter** [comment]
Can you point me to the repo?

**Comment by Dockter** [comment]
FYI, my wife reported that the .zip file you linked is missing the particle affect in the model.

**Comment by Bysokar** [comment]
@Dockter https://github.com/stargatemc/sgcraft-1.12.2
Particle effect? Maybe that was what was lagging it?

**Comment by Bysokar** [comment]
@Dockter Also, apologies in advance if the code is a mess and for my commit comments. Im self taught java & I've been doing things solo and didn't really need to bother with them :)

**Comment by Dockter** [comment]
Is that repo hidden?  I can't see it via the above link.

**Comment by Bysokar** [comment]
@Dockter your invited as a collaborator (before you  sent that message)… try visiting https://github.com/stargatemc
Yes its hidden. Our modifications are mostly for it to integrate with our coremod/advanced rocketry. 
Please don't share it :)

**Comment by Dockter** [comment]
I did not get an invite, I still cannot see that repo.

**Comment by Bysokar** [comment]
![image](https://user-images.githubusercontent.com/66674931/88244607-a23a8300-ccd7-11ea-867f-f857edaac1b8.png)
Deleted and re-added the invite..... Maybe the link is case sensitive?

**Comment by C4PTRAMPAGE** [comment]
can confirm that the 2.0.5 update seems to completely fix the lag problem, haven't done as much testing as i could though

**Comment by Dockter** [comment]
> [@Dockter](https://github.com/Dockter) https://github.com/stargatemc/sgcraft-1.12.2 Particle effect? Maybe that was what was lagging it?

Did this repo get deleted?
